redfish-core/lib/power_supply_metrics.hpp

```diff
@@ -0,0 +1,248 @@
+
+    crow::connections::systemBus->async_method_call(
+        [aResp, powerSupplyID, maximumPath](
+            const boost::system::error_code ec,
+            const std::vector<std::pair<std::string, std::vector<std::string>>>&
+                object) {
+            if (ec)
+            {
+                BMCWEB_LOG_DEBUG << "DBUS response error";
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            for (const auto& [serviceName, interfaceList] : object)
+            {
+                for (const auto& interface : interfaceList)
+                {
+                    if (interface ==
+                        "org.open_power.Sensor.Aggregation.History.Maximum")
+                    {
+                        crow::connections::systemBus->async_method_call(
+                            [aResp, powerSupplyID](
+                                const boost::system::error_code ec2,
+                                const std::variant<std::vector<
+                                    std::tuple<std::uint64_t, std::int64_t>>>&
+                                    values_rsp) {
+                                if (ec2)
+                                {
+                                    BMCWEB_LOG_DEBUG << "DBUS response error";
+                                    messages::internalError(aResp->res);
+                                    return;
+                                }
+
+                                const std::vector<
+                                    std::tuple<std::uint64_t, std::int64_t>>*
+                                    values_ptr =
+                                        std::get_if<std::vector<std::tuple<
+                                            std::uint64_t, std::int64_t>>>(
+                                            &values_rsp);
+                                if (values_ptr != nullptr)
+                                {
+                                    nlohmann::json& maximums =
+                                        aResp->res.jsonValue["Max"];
+                                    for (const auto& values : *values_ptr)
+                                    {
+                                        // The first value returned is the
+                                        // timestamp, already handled in
+                                        // getAverageValues().
+                                        // The second value returned is in
+                                        // watts. It is the maximum watts this
+                                        // power supply has used in a 30 second
+                                        // interval.
+                                        maximums.push_back(std::get<1>(values));
+                                    }
+                                }
+                                else
+                                {
+                                    BMCWEB_LOG_DEBUG
```

Comment:
You could increase this to ERROR if you want

Reply:
Ah! Yes, that probably makes more sense.

Is there a reason why "DBUS response error" is DEBUG and not ERROR? (I copy/pasted what others were doing).